lib/parse-client-options.js

```diff
@@ -38,18 +55,22 @@ function parseOptions (userOptions) {
     }
   }
 
-  clientDefaults.baseUrl = `${options.protocol}://${options.host}`
+  if (options.baseUrl) {
+    clientDefaults.baseUrl = `${options.baseUrl}`
+  } else {
+    clientDefaults.baseUrl = `${options.protocol}://${options.host}`
 
-  if (options.port) {
-    clientDefaults.baseUrl += `:${options.port}`
-  }
+    if (options.port) {
+      clientDefaults.baseUrl += `:${options.port}`
+    }
 
-  // Check if a prefix is passed in the options and strip any leading or trailing slashes from it.
-  if (options.pathPrefix) {
-    clientDefaults.baseUrl += '/' + options.pathPrefix.replace(/(^[/]+|[/]+$)/g, '')
+    // Check if a prefix is passed in the options and strip any leading or trailing slashes from it.
+    if (options.pathPrefix) {
+      clientDefaults.baseUrl += '/' + options.pathPrefix.replace(/(^[/]+|[/]+$)/g, '')
+    }
   }
-
   /* istanbul ignore else */
+
```

Comment:
let’s implement this a little different. First, let’s set `clientDefaults.baseUrl` directly in line 51

```js
  const clientDefaults = {
    baseUrl: options.baseUrl,
    headers: options.headers,
    request: {
      timeout: options.timeout
    }
  }
```

Because that’s the default path now, and once we remove the deprecated options, we can just leave it as is.

Next instead of

```js
  if (options.baseUrl) {
    clientDefaults.baseUrl = `${options.baseUrl}`
  } else {
```

I would suggest to check for `if (userOptions.host) { ...` because the host options must always be set. Then within the if block, replace `options.*` with `userOptions.*`, because we directly want to check for what the user passed, without the default values applied to them and without being filtered by `OPTION_NAMES` defined on top.